### DISCUSSION TITLE
Append layout / workspace name / escaping

### DISCUSSION THREAD

**Discussion by cnapieral** [i3/i3#5748] [Q&A]
Hi everyone,
i need help with my i3wm config. Goal is to have terminals opening on workspace 1 on startup.
I used save-tree as mentioned in the docs to create json-layout.

My workspace is named the following:
set $ws1 "1: >_"

I tried with the following command:

3-msg "1: \\>\\_; append_layout /home/lala/.i3/workspace-1.json"

And other variants of escaping but it does not work, its always try to interpret ">". When i rename my workspace to eg. "workspace 1" it works. Any ideas? How do you guys manage?

Best regards,
Chris

edit:
edited escaping as github escapes :D

edit 2:
i changed ws1 to test.
i also tried replacing "workspace 1" with the variable $ws1, but the output is:
exec --no-startup-id "i3-msg 'workspace '"test"'; append_layout /home/lala/.i3/workspace-1.json'"

so i have extra "" around the variable and cannot get rid of them.

edit 3: 
guys, FORGET everything. I was a total dumb a**. I was to focused on the wrong way. After just using the workspace name as it is (including special characters), it is working now. Topic can be closed.

**Comment by Mazoch** [comment]
I'm not entirely sure if this is what you mean, but I try to help you. When I want to open terminal in specific workspace and autmatically go to it, I use this command in i3 config file.

<pre><code>assign [class=(?i)kitty] $ws1
for_window [class=(?i)kitty] focus</pre></code>

**Reply by cnapieral** [reply]
Hi mazoch and thanks for your reply, but thats not what i was looking for. 

It is about restoring a layout. Eg on ws1 you have 3 terminals in a specific layout (2 containers vertical splitted and 1 subcontainer horizontal splitted). I now want it to restore with i3-msg and append_layout.
